### PR TITLE
Comments on review should have the same sha

### PR DESCRIPTION
**Pull request by zeripath**
Restore #13217

### CODE REVIEW

**Comment by zeripath** [comment]
Whilst the migration was broken the correct solution should have been to no-op the migration not revert the whole commit.

In any case hopefully this fixes the migration.

**Review by lafriks — APPROVED** [review]

**Comment by lunny** [comment]
CI randomly failed but may not relate to this.

**Comment by zeripath** [comment]
Unfortunately the mssql code is incorrect

**Comment by zeripath** [comment]
mssql: Only one expression can be specified in the select list when the subquery is not introduced with EXISTS.

File: models/migrations/v158.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+		}
+
+		start += len(comments)
+
+		if len(comments) < batchSize {
+			break
+		}
+	}
+
+	return sess.Commit()
```
Comment (lunny) [inline models/migrations/v158.go]: The transaction may too big to execute failure.

Comment (zeripath) [reply models/migrations/v158.go]: OK I've moved the session within the batch

Comment (lunny) [reply models/migrations/v158.go]: It's better to move new session out of `for`. A xorm session could run multiple transactions on the same go routine.

Comment (zeripath) [reply models/migrations/v158.go]: done

**Review by lunny — APPROVED** [review]

**Comment by techknowlogick** [comment]
🚀

**Comment by jpraet** [comment]
This may have caused a regression in 1.13 #13683. Could someone have a look at that?

**Pull request merged**